services/src/ogc/util.rs

```diff
@@ -280,9 +303,28 @@ pub fn ogc_endpoint_url(base: &Url, protocol: OgcProtocol, workflow: WorkflowId)
         .map_err(Into::into)
 }
 
+pub struct OgcRequestGuard<'a> {
+    request: &'a str,
+}
+
+impl<'a> OgcRequestGuard<'a> {
+    pub fn new(request: &'a str) -> Self {
+        Self { request }
+    }
+}
+
+impl Guard for OgcRequestGuard<'_> {
+    fn check(&self, ctx: &GuardContext<'_>) -> bool {
+        ctx.head().uri.query().map_or(false, |q| {
+            q.contains(&format!("request={}", self.request))
+                || q.contains(&format!("REQUEST={}", self.request))
```

Comment:
why not case insensitive?

Reply:
I can do it case insensitive here if you want but it would be inconsistent because all the other parameters are also only supported all/no caps. Also that would require a regex I guess.

Reply:
What about lowercasing the whole string and then using contains?

Reply:
AFAIK GetCapabilities etc. have to be in camelCase.